fraud-detection-new-isoscape/origin_validation.py

```diff
@@ -50,103 +49,73 @@ def get_asset_list(parent_name) -> list:
-
-        """**Nitrogen**
 
-    Isoscape
-    """
-
-        isoscape_mean = self.nitrogen_means_iso.reduceRegion(
+    
+    def _calc_p_value(
+            measured_mean,
+            measured_std,
+            num_measurements,
+            poi, 
+            isoscape):
+        isoscape_mean = isoscape.select('b1').reduceRegion(
             reducer=ee.Reducer.mean(),
-            geometry=self.poi,
+            geometry=poi,
             scale=30).getInfo().get('b1')
-        isoscape_var = self.nitrogen_variances_iso.reduceRegion(
+        isoscape_var = isoscape.select('b2').reduceRegion(
             reducer=ee.Reducer.mean(),
-            geometry=self.poi,
+            geometry=poi,
             scale=30).getInfo().get('b1')
-        if isoscape_var is not None:
-            isoscape_std = np.sqrt(isoscape_var)
-        else:
-            isoscape_std = 0
-
-        """Isotope calculation"""
-
-        isotope_mean = np.mean(self.nitrogen_measurements)
-        isotope_std = np.std(self.nitrogen_measurements)
-        #print(isotope_mean, isotope_std, isoscape_mean, isoscape_std)
+        isoscape_std = np.sqrt(isoscape_var)
+    
+        _, p_value = scipy.stats.ttest_ind_from_stats(
+            measured_mean, measured_std, 
+            num_measurements, isoscape_mean,
+            isoscape_std, 30, 
+            equal_var=False, alternative="two-sided")
 
-        """ttest and p-value"""
+        return p_value
 
-        _, p_value_nitrogen = scipy.stats.ttest_ind_from_stats(
-            isotope_mean, isotope_std, len(self.nitrogen_measurements), isoscape_mean, isoscape_std, 30, equal_var=False, alternative="two-sided")
-
-        """Origin Verification"""
-
-        origin_validity = p_value_oxygen * p_value_carbon * p_value_nitrogen
-        return origin_validity, p_value_oxygen, p_value_carbon, p_value_nitrogen
+    def evaluate(self):
+        p_value_oxygen, p_value_carbon, p_value_nitrogen = 1, 1, 1
+        if self.oxygen_isoscape:
+            p_value_oxygen = _calc_p_value(
+                np.mean(self.oxygen_measurements),
+                np.std(self.oxygen_measurements),
+                len(self.oxygen_measurements),
+                self.poi,
+                self.oxygen_isoscape)
+        
+        if self.carbon_isoscape:
+            p_value_carbon = _calc_p_value(
+                np.mean(self.carbon_measurements),
+                np.std(self.carbon_measurements),
+                len(self.carbon_measurements),
+                self.poi,
+                self.carbon_isoscape)
+        
+        if self.nitrogen_isoscape:
+            p_value_nitrogen = _calc_p_value(
+                np.mean(self.nitrogen_measurements),
+                np.std(self.nitrogen_measurements),
+                len(self.nitrogen_measurements),
+                self.poi,
+                self.nitrogen_measurements)
+        
+        combined_p_value = p_value_oxygen * p_value_carbon * p_value_nitrogen
```

Comment:
Yes, by default I'm leaving the p-values of the other elements as 1.0 if those isoscapes don't exist so they don't affect the combined p_value.

Reply:
but if someone accidentally uploads a C or N isoscape, it will change the combined p-values?  This is probably a bad idea and falls into the category of creating a trap for a future dev.
Let's just add a comment about explicitly only including O.

Reply:
Ok, i added 3 flags:

```
_ENABLE_d18O_ANALYSIS = True
_ENABLE_d13C_ANALYSIS = False
_ENABLE_d15N_ANALYSIS = False
```

And block the alteration of the p_values if these are false. Is that good?

Reply:
I dig it.  Just a few nits -- comments and constants.  Feel free to ping on those and will approve